core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletMetadata.java

```diff
@@ -156,8 +156,13 @@ private TabletMetadata(Builder tmBuilder) {
     this.compacted = tmBuilder.compacted.build();
     this.userCompactionsRequested = tmBuilder.userCompactionsRequested.build();
     this.unSplittableMetadata = tmBuilder.unSplittableMetadata;
-    this.fileSize =
-        Suppliers.memoize(() -> files.values().stream().mapToLong(DataFileValue::getSize).sum());
+    this.fileSize = Suppliers.memoize(() -> {
```

Comment:
`files.values().stream().mapToLong(DataFileValue::getSize).sum()` has the following call stack:

  * [values](https://github.com/openjdk/jdk/blob/jdk-11%2B28/src/java.base/share/classes/java/util/AbstractMap.java#L405) (creates new AbstractCollection)
  * [stream](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/Collection.java#L747)  (calls spliterator and then StreamSupport.stream)
    * [spliterator](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/Collection.java#L728)  (calls Spliterators.spliterator)
      * [Spliterators.spliterator](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/Spliterators.java#L140) (creates new ArraySpliterator)
    * [StreamSupport.stream](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/stream/StreamSupport.java#L67) (returns a new ReferencePipeline.Head object)
  * [mapToLong](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/stream/ReferencePipeline.java#L240) ( creates a new LongPipeline.StatelessOp object)
  * [sum](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/stream/LongPipeline.java#L458) (calls reduce())
    * [reduce](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/stream/LongPipeline.java#L501 (calls ReduceOps.makeLong and then evaluate)
      * [ReduceOps.makeLong](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/stream/ReduceOps.java#L462) (creates a new ReduceOp and returns it)
      *   [evaluate](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/stream/AbstractPipeline.java#L257)

Reply:
Not exactly sure why streams are so much slower, its probably the long call stack and object allocation as the stream works through each item that @dlmarion mentioned.  Realizing its probably best to avoid streams for cases where code is called really frequently.  We saw this in accumulo-access where we have a jhm performance benchmark test, replacing a stream with a for loop caused a massive increase in performance for code that was executed per expression.  Was a bit surprised at the difference that small change made in performance for accumulo-access.

Reply:
Added a comment in https://github.com/apache/accumulo/commit/f2fdf0b010bc7203bdf3cf13a0ee17a0cca48356